File: Part 2 - Regression/Section 4 - Simple Linear Regression/R/my_simple_linear_regression.R
# 1. Data Preprocessing

# A| Importing Data
dataset = read.csv('Salary_Data.csv')

# B| Splitting Data
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# 2. Simple Linear Regression

# A| Fitting Model into Training Set
regressor = lm(formula = Salary ~ YearsExperience, data = training_set)

# B| Predicting Results
y_pred = predict(regressor, newdata = test_set)

# C| Visualizing Models
# install.packages('ggplot2')
library(ggplot2)

# Training Set
ggplot() +
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary),
             color='red') +
  geom_line(aes(x= training_set$YearsExperience, y= predict(regressor, newdata = training_set)),
            color='blue') +
  ggtitle('Salary vs Experience (Training Set)') +
  xlab('Years of Experience') +
  ylab('Salary')


# Test Set
ggplot() +
  geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary),
             color='green') +
  geom_line(aes(x= training_set$YearsExperience, y= predict(regressor, newdata = training_set)),
            color='blue') +
  ggtitle('Salary vs Experience (Training Set)') +
  xlab('Years of Experience') +
  ylab('Salary')

File: Part 2 - Regression/Section 4 - Simple Linear Regression/Python/my_simple_linear_regression.py
# 1. Data Preprocessing

# A| Import Libraries
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd

# B| Import Data
dataset = pd.read_csv('Salary_Data.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:,  -1].values

# C| Handling Missing Stuff
from sklearn.preprocessing import Imputer
imputer = Imputer()
imputer = imputer.fit(X)
X = imputer.transform(X)

# D| Encoding Categorical Data - no use here

# E| Splitting Data
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# F| Feature Scaling - no use here

# --------------------------------- #



# 2. Simple Linear Regression

# A| Fitting Simple Linear Model to Training Set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train,y_train)

# B| Predicting the Test Set Results
y_pred = regressor.predict(X_test)

# C| Visualising Training Set Results
plt.scatter(X_train, y_train, color='red', edgecolors='black', alpha=0.75)
plt.plot(X_train, regressor.predict(X_train), color='blue')
plt.title('Salary vs Experience (Training Set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()

# D| Visualising Test Set Results
plt.scatter(X_test, y_test, edgecolors='black', color='green', alpha=0.75)
plt.plot(X_train, regressor.predict(X_train), color='orange')
plt.title('Salary vs Experience (Test Set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()
